src/main/java/com/nimbusds/jose/crypto/ECDH1PUDecrypter.java

```diff
@@ -0,0 +1,275 @@
+ * </ul>
+ *
+ * @author Alexander Martynov
+ * @version 2021-08-03
+ */
+@ThreadSafe
+public class ECDH1PUDecrypter extends ECDH1PUCryptoProvider implements JWEDecrypter, CriticalHeaderParamsAware {
+
+
+	/**
+	 * The supported EC JWK curves by the ECDH crypto provider class.
+	 */
+	public static final Set<Curve> SUPPORTED_ELLIPTIC_CURVES;
+
+
+	static {
+		Set<Curve> curves = new LinkedHashSet<>();
+		curves.add(Curve.P_256);
+		curves.add(Curve.P_384);
+		curves.add(Curve.P_521);
+		SUPPORTED_ELLIPTIC_CURVES = Collections.unmodifiableSet(curves);
+	}
+
+
+	/**
+	 * The private EC key.
+	 */
+	private final ECPrivateKey privateKey;
+
+	/**
+	 * The public EC key.
+	 */
+	private final ECPublicKey publicKey;
+
+	/**
+	 * The critical header policy.
+	 */
+	private final CriticalHeaderParamsDeferral critPolicy = new CriticalHeaderParamsDeferral();
+
+
+	/**
+	 * Creates a new Elliptic Curve Diffie-Hellman decrypter.
+	 *
+	 * @param privateKey 	The private EC key. Must not be {@code null}.
+	 * @param publicKey 	The public EC key. Must not be {@code null}.
+	 *
+	 * @throws JOSEException If the elliptic curve is not supported.
+	 */
+	public ECDH1PUDecrypter(final ECPrivateKey privateKey,
+							final ECPublicKey publicKey)
+		throws JOSEException {
+
+		this(privateKey, publicKey, null);
+	}
+
+	/**
+	 * Creates a new Elliptic Curve Diffie-Hellman decrypter.
+	 *
+	 * @param privateKey     The private EC key. Must not be {@code null}.
+	 * @param publicKey 	 The public EC key. Must not be {@code null}.
+	 * @param defCritHeaders The names of the critical header parameters
+	 *                       that are deferred to the application for
+	 *                       processing, empty set or {@code null} if none.
+	 *
+	 * @throws JOSEException If the elliptic curve is not supported.
+	 */
+	public ECDH1PUDecrypter(final ECPrivateKey privateKey,
+							final ECPublicKey publicKey,
+							final Set<String> defCritHeaders)
+		throws JOSEException {
+
+		this(privateKey, publicKey, defCritHeaders, Curve.forECParameterSpec(privateKey.getParams()));
+	}
+
+
+	/**
+	 * Creates a new Elliptic Curve Diffie-Hellman decrypter. This
+	 * constructor can also accept a private EC key located in a PKCS#11
+	 * store that doesn't expose the private key parameters (such as a
+	 * smart card or HSM).
+	 *
+	 * @param privateKey     The private EC key. Must not be {@code null}.
+	 * @param publicKey 	 The public EC key. Must not be {@code null}.
+	 * @param defCritHeaders The names of the critical header parameters
+	 *                       that are deferred to the application for
+	 *                       processing, empty set or {@code null} if none.
+	 * @param curve          The key curve. Must not be {@code null}.
+	 *
+	 * @throws JOSEException If the elliptic curve is not supported.
+	 */
+	public ECDH1PUDecrypter(final ECPrivateKey privateKey,
+							final ECPublicKey publicKey,
+                            final Set<String> defCritHeaders,
```

Comment:
Please, consistently use spaces or tabs, but not both.
What are the code formatting guidelines for the Nimbus lib?

Reply:
They use both tabs and spaces

Reply:
I left spaces for my files